web/genericcom.js

```diff
@@ -70,20 +70,53 @@ class MLManager {
 }
 
 class FakeMLManager {
-  constructor({ enableGuessAltText }) {
+  eventBus = null;
+
+  constructor({ enableGuessAltText, enableAltTextModelDownload }) {
     this.enableGuessAltText = enableGuessAltText;
+    this.enableAltTextModelDownload = enableAltTextModelDownload;
+    this.hasProgress = false;
   }
 
   async isEnabledFor(_name) {
     return this.enableGuessAltText;
   }
 
   async deleteModel(_service) {
+    this.enableAltTextModelDownload = false;
     return null;
   }
 
+  async downloadModel(_name) {
```

Comment:
Quickly testing the preview, it's not clear to me if this method actually works since nothing seems to happen when trying to trigger downloading!?

Also, this method isn't a wonder of readability :-)
At least a basic comment would be helpful, or perhaps refactor things slightly?

Reply:
To test it:
 - delete the model;
 - download it;
 - close settings;
 - add an image;
 - you should see the download progress in new alt-text dialog.